torchvision/prototype/features/_dataset_wrapper.py

```diff
@@ -0,0 +1,388 @@
+    child_pytrees, context = flatten_fn(pytree)
+
+    flat = []
+    for children_pytree, children_spec in zip(child_pytrees, spec.children_specs):
+        flat.extend(tree_flatten_to_spec(children_pytree, children_spec))
+
+    return flat
+
+
+class VisionDatasetFeatureWrapper(Dataset):
+    _wrappers_fns: Dict[Type[datasets.VisionDataset], PyTree] = {}
+
+    def __init__(self, dataset: datasets.VisionDataset, wrappers: PyTree) -> None:
+        self.vision_dataset = dataset
+        self.sample_wrappers = wrappers
+
+        # We need to disable the transforms on the dataset here to be able to inject the wrapping before we apply the
+        # transforms
+        self.transform, dataset.transform = dataset.transform, None
+        self.target_transform, dataset.target_transform = dataset.target_transform, None
+        self.transforms, dataset.transforms = dataset.transforms, None
+
+    def __getattr__(self, item: str) -> Any:
+        with contextlib.suppress(AttributeError):
+            return object.__getattribute__(self, item)
+
+        return getattr(self.vision_dataset, item)
+
+    def __getitem__(self, idx: int) -> Any:
+        # This gets us the raw sample since we disabled the transforms for the underlying dataset in the constructor
+        # of this class
+        sample = self.vision_dataset[idx]
+
+        wrappers_flat, spec = tree_flatten(self.sample_wrappers)
+        # We cannot use `tree_flatten` directly, because the spec of `self.sample_wrappers` and `sample` might differ.
+        # For example, for `COCODetection` the target is a list of dicts. To be able to wrap this into a dict of lists,
+        # we need to have access to the whole list, but `tree_flatten` would also flatten it.
+        sample_flat = tree_flatten_to_spec(sample, spec)
+        wrapped_sample_flat = [wrapper(item) for item, wrapper in zip(sample_flat, wrappers_flat)]
+        sample = tree_unflatten(wrapped_sample_flat, spec)
+
+        # We don't need to care about `transform` and `target_transform` here since `VisionDataset` joins them into a
+        # `transforms` internally:
+        # https://github.com/pytorch/vision/blob/2d92728341bbd3dc1e0f1e86c6a436049bbb3403/torchvision/datasets/vision.py#L52-L54
+        if self.transforms is not None:
+            sample = self.transforms(*sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        return len(self.vision_dataset)
+
+    @classmethod
+    def _register_wrappers_fn(cls, dataset_type: Type[datasets.VisionDataset]) -> Callable[[PyTree], PyTree]:
+        def register(wrappers_fn: PyTree) -> PyTree:
+            cls._wrappers_fns[dataset_type] = wrappers_fn
+            return wrappers_fn
+
+        return register
+
+    @classmethod
+    def from_torchvision_dataset(
+        cls,
+        dataset: datasets.VisionDataset,
+        *,
+        keep_pil_image: bool = False,
+        bounding_box_format: Optional[features.BoundingBoxFormat] = None,
```

Comment:
This parameter gives users the option to select the format. It could be useful so users always get bounding boxes in the "right" format for them. However, we have a 

https://github.com/pytorch/vision/blob/b8af91ab78d9e95ac377e059c7c47249128d5bdb/torchvision/prototype/transforms/_meta.py#L10

transform that does the same. I've not made up my mind yet.

Reply:
It does look a bit redundant with the transforms.

Perhaps I'm pushing it a little, but: why have a `bounding_box_format` but not a `image_size` (assuming we agree `image_size` isn't something we want :) ! )

Reply:
> Perhaps I'm pushing it a little, but: why have a `bounding_box_format` but not a `image_size` (assuming we agree `image_size` isn't something we want :) ! )

I guess you mean `spatial_size`

https://github.com/pytorch/vision/blob/5dd95944c609ac399743fa843ddb7b83780512b3/torchvision/prototype/datapoints/_bounding_box.py#L18-L20

right? If so, we can't make this a free parameter, because it is tied to the image in the same sample. The format is not tied to anything.

That being said, we no longer need this. I'm not sure what the state was when I wrote the comment, but the API is now agnostic with regards to bounding box format. Plus, if the user actually wants to force a specific format, they can drop a

https://github.com/pytorch/vision/blob/5dd95944c609ac399743fa843ddb7b83780512b3/torchvision/prototype/transforms/_meta.py#L11

into their transforms pipeline.